internal/config_checks_avr.h

```diff
@@ -133,6 +133,8 @@
 #define MOZZI_AUDIO_BITS (2*MOZZI_AUDIO_BITS_PER_CHANNEL)
 #endif
 
+#define MOZZI__INTERNAL_ANALOG_READ_RESOLUTION 10
```

Comment:
One thing is that, on some platform, the ADC resolution is editable (`analogReadResolution()`) maybe this define should be editable like other Mozzi's config at the top of the user sketch. In that case, a user can let mozzi knows of the actual ADC resolution? It would also solves partly the problem of mbed, where all platforms might not have the same resolution?

Reply:
I think the way to implement that will be:
- user defines MOZZI_ANALOG_READ_RESOLUTION
- implementation can see this, and may decide to set the internal resolution, accordingly. Approximately:

```
#ifdef MOZZI_ANALOG_READ_RESOLUTION
#define MOZZI__INTERNAL_ANALOG_READ_RESOLUTION MOZZI_ANALOG_READ_RESOLUTION
#define MOZZI__IMPL_SET_ANALOG_READ_RESOLUTION
#endif

[...]

//inside setupMozziADC
#ifdef MOZZI__IMPL_SET_ANALOG_READ_RESOLUTION
analogReadResolution(MOZZI_ANALOG_READ_RESOLUTION);
#endif
```

The advantages are that only one define is needed, implementations are free to set this internally, or rely solely on shifting, and users won't need to worry about which it is.

Reply:
Oh, that is even one step further than what I though (I was thinking that the user sets `analogReadResolution` in the user code and let Mozzi know about it with the define). I guess the `analogReadResolution` should be portable so probably even better to put it there!

Reply:
Note that we may or may not take readings at the ```analogReadResolution()```, as we may or may not utilize the same mechanisms as ```analogRead()```. The pico, for instance does not actually seem to support requesting anything other than 12 bits from the hardware, and ```analogRead()``` merely shifts that at runtime. As we're not using that, we always need to shift from 12 bits, irrespective of analogReadResolution(). I _believe_ on AVR, we're similarly working at a fixed resolution, internally.

STM32(duino) and Mbed (via Arduino_AdvancedAnalog) do seem to support hardware reads at different resolution (again independent from ```analogReadResolution()```), although not all bit values are possible (no 13 bits, for instance), and I'm a bit reluctant about touching the already complex code, here.

So what I've outlined above is a really just theory for the time being. I'll add a note to that effect in the template files, though.